recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,444 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import re
+
+
+class CPythonConan(ConanFile):
+    name = "cpython"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.python.org"
+    description = "Python is a programming language that lets you work quickly and integrate systems more effectively."
+    topics = ("conan", "python", "cpython", "language", "script")
+    license = ("Python-2.0",)
+    exports_sources = "patches/**"
+    generators = "visual_studio"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "optimizations": [True, False],
+        "lto": [True, False],
+        "docstrings": [True, False],
+        "pymalloc": [True, False],
+        "with_bz2": [True, False],
+        "with_gdbm": [True, False],
+        "with_nis": [True, False],
+        "with_sqlite3": [True, False],
+        "with_tkinter": [True, False],
+        # "with_curses": [True, False],
+
+        # Python 2 options
+        "unicode": ["ucs2", "ucs4"],
+        "with_bsddb": [True, False],
+        # Python 3 options
+        "with_lzma": [True, False],
+
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "optimizations": False,
+        "lto": False,
+        "docstrings": True,
+        "pymalloc": True,
+        "with_bz2": True,
+        "with_gdbm": True,
+        "with_nis": False,
+        "with_sqlite3": True,
+        "with_tkinter": False,
+        # "with_curses": False,
+
+        # Python 2 options
+        "unicode": "ucs2",
+        "with_bsddb": True,
+        # Python 3 options
+        "with_lzma": True,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _version_major_minor(self):
+        if self.settings.compiler == "Visual Studio":
+            joiner = ""
+        else:
+            joiner = "."
+        return joiner.join(self.version.split(".")[:2])
+
+    @property
+    def _is_py3(self):
+        return self.version.split(".")[0] == "3"
+
+    @property
+    def _is_py2(self):
+        return self.version.split(".")[0] == "2"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            del self.options.lto
+            del self.options.docstrings
+            del self.options.pymalloc
+            del self.options.with_gdbm
+            del self.options.with_nis
+            del self.options.with_curses
```

Comment:
comment out until curses option is working

Reply:
I've added the option back, but have it disabled by default (for now).